frontend/packages/console-app/src/components/pdb/PDBFormPage.tsx

```diff
@@ -66,9 +67,18 @@ export const PDBFormPage: React.FC<{
     selector: {},
   };
   const [helpText, setHelpText] = React.useState(formHelpText);
-  const k8sObj = pdbToK8sResource(initialPDB);
+  const k8sObj = pdbToK8sResource(initialPDB, existingResource);
+
   const YAMLEditor: React.FC<YAMLEditorProps> = ({ onChange, initialYAML = '' }) => {
-    return <CreateYAML hideHeader match={match} onChange={onChange} template={initialYAML} />;
+    return (
+      <CreateYAML
+        hideHeader
+        match={match}
+        onChange={onChange}
+        template={initialYAML || safeJSToYAML(existingResource)}
```

Comment:
My guess is your `setYAML` to empty string was for this use-case. Wouldn't you want to just use `existingResource` always if it's present?

```suggestion
        template={existingResource ? safeJSToYAML(existingResource) : initialYAML}
```

Upon further thought and testing of the code ... I think you're trying to get around a problem with the UI is not fully loaded yet. What you want to do is check on the loading state of the K8s calls (2nd param in the array returned by `useK8sWatchResource`) and just delay rendering the switcher...

```
  const [resource, loadedResource] = useK8sWatchResource<K8sPodControllerKind>({
. . .
  const [pdbResources, loadedPDBResource] = useK8sWatchResource<PodDisruptionBudgetKind[]>({
. . .
  const stillLoading = !loadedResource || !loadedPDBResource;

  return (
    <>
      . . .
      {stillLoading ? (
        <LoadingBox />
      ) : (
        <SyncedEditor . . . />
      )}
      . . .
```

Reply:
If you need the 'latest' -- how did this work? You were using the initialYAML somehow to keep things up to date?